src/modules/75-ce/components/COAsgSelector/index.tsx

```diff
@@ -186,4 +246,101 @@ const COAsgSelector: React.FC<COAsgSelectorprops> = props => {
   )
 }
 
+interface GCPFiltersProps {
+  region?: SelectOption
+  zone?: SelectOption | null
+}
+
+interface GroupsFilterProps {
+  gatewayDetails: GatewayDetails
+  onGcpFiltersChangeCallback: (values: GCPFiltersProps, loading: boolean) => void
+  isEditFlow: boolean
+}
+
+const GroupsFilter: React.FC<GroupsFilterProps> = ({ gatewayDetails, onGcpFiltersChangeCallback, isEditFlow }) => {
+  const { accountId } = useParams<AccountPathProps>()
+  const { getString } = useStrings()
+  const isGcpProvider = Utils.isProviderGcp(gatewayDetails.provider)
+
+  // GCP filters data
+  const { data: regionsData, loading: regionsLoading } = useRegionsForSelection({
+    cloudAccountId: gatewayDetails.cloudAccount.id,
+    additionalProps: {}
+  })
+  const [selectedRegion, setSelectedRegion] = useState<SelectOption>()
+  const [zonesData, setZonesData] = useState<SelectOption[]>([])
+  const [selectedZone, setSelectedZone] = useState<SelectOption | null | undefined>()
+
+  const {
+    data: zones,
+    loading: zonesLoading,
+    refetch: fetchZones
+  } = useAllZones({
+    account_id: accountId,
+    queryParams: {
+      cloud_account_id: gatewayDetails.cloudAccount.id,
+      accountIdentifier: accountId,
+      region: ''
+    },
+    lazy: true
+  })
+
+  useEffect(() => {
+    if (selectedRegion) {
+      fetchZones({
+        queryParams: {
+          cloud_account_id: gatewayDetails.cloudAccount.id,
+          accountIdentifier: accountId,
+          region: selectedRegion.label
+        }
+      })
+    }
+    if (isGcpProvider) {
+      setSelectedZone(null)
+      onGcpFiltersChangeCallback({ region: selectedRegion }, regionsLoading)
+    }
+  }, [selectedRegion, regionsLoading])
```

Comment:
Codacy has a fix for the issue: [React Hook useEffect has missing dependencies: 'accountId', 'fetchZones', 'gatewayDetails.cloudAccount.id', 'isGcpProvider', and 'onGcpFiltersChangeCallback'. Either include them or remove the dependency array. If 'onGcpFiltersChangeCallback' changes too often, find the parent component that defines it and wrap that definition in useCallback.](https://app.codacy.com/gh/harness/harness-core-ui/pullRequest?prid=9232281)
```suggestion
  }, [selectedRegion, regionsLoading, isGcpProvider, fetchZones, gatewayDetails.cloudAccount.id, accountId, onGcpFiltersChangeCallback])
```